File: src/Flowsharp.Abstractions/Activities/Workflow.cs
using System.Collections.Generic;
using System.Linq;
using Flowsharp.ActivityResults;
using Flowsharp.Models;

namespace Flowsharp.Activities
{
    public class Workflow
    {
        public Workflow()
        {
        }

        public Workflow(IEnumerable<IActivity> activities, IEnumerable<Connection> connections)
        {
            Activities = activities.ToList();
            Connections = connections.ToList();    
        }
        
        public IList<IActivity> Activities { get; set; } = new List<IActivity>();
        public IList<Connection> Connections { get; set; } = new List<Connection>();
    }
}

File: src/web/modules/Flowsharp.Web.ViewComponents/Assets/scripts/flowsharp/workflow.ts
///<reference path="activity.ts"/>
///<reference path="connection.ts"/>
///<reference path="workflow-metadata.ts"/>
///<reference path="workflow-status.ts"/>

namespace Flowsharp {
    export interface IWorkflow {
        metadata: IWorkflowMetadata;
        activities: IActivity[];
        connections: IConnection[];
        status: WorkflowStatus;
    }   
}